### PR TITLE
Feat/simplifies naming convention writing

### PR DESCRIPTION
**Pull request by rudolfix**
### Description
A followup on #998 
- adds examples for custom naming conventions
- simplifies how naming conventions are implemented and configured
- adds some tests

### CODE REVIEW

File: docs/examples/custom_naming/sql_ci_no_collision.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
+from typing import ClassVar
```
Comment (sh-rp) [inline docs/examples/custom_naming/sql_ci_no_collision.py:1]: these extra files will not show up in the docs, maybe this is ok for now, but we might need to adapt this somehow?

File: docs/examples/custom_naming/custom_naming.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+With this example you will learn to:
+* Create a naming convention module with a recommended layout
+* Use naming convention by explicitly passing it to `duckdb` destination factory
+* Use naming convention by configuring it config.toml
+* Changing the declared case sensitivity by overriding `is_case_sensitive` property
+* Providing custom normalization logic by overriding `normalize_identifer` method
+
+"""
+
+import dlt
```
Comment (sh-rp) [inline docs/examples/custom_naming/custom_naming.py:28]: really cool example!

File: docs/tools/prepare_examples_tests.py [hunk 3]
```diff
@@ -54,8 +54,12 @@
             os.unlink(test_example_file)
             continue
 
-        with open(example_file, "r", encoding="utf-8") as f:
-            lines = f.read().split("\n")
+        try:
+            with open(example_file, "r", encoding="utf-8") as f:
+                lines = f.read().split("\n")
+        except FileNotFoundError:
```
Comment (sh-rp) [inline docs/tools/prepare_examples_tests.py:60]: what is this for? when does this happen?

File: docs/website/docs/general-usage/naming-convention.md [hunk 4]
```diff
@@ -66,16 +58,20 @@ In the snippet below, we define a resource with various "illegal" unicode charac
...
-* `dlt` applies final naming convention in `normalize` step. Naming convention comes from (1) explicit configuration (2) from destination capabilities. Naming convention
-in schema will be ignored.
-* You can change the naming convention in the capabilities: (name, case-folding, case sensitivity)
+* Each destination may define a preferred naming convention (ie. Weaviate), otherwise **snake case** will be used
+* This naming convention is used when new schemas are created. This happens when pipeline is run for a first time.
+* Schemas preserve naming convention when saved. Your running pipelines will maintain existing naming conventions if not requested otherwise
+* `dlt` applies final naming convention in `normalize` step. Naming convention comes from (1) explicit configuration (2) from destination capabilities.
+* Naming convention will be used to put destination is case sensitive/insensitive mode and apply the right case folding function.
+
+:::caution
```
Comment (sh-rp) [inline docs/website/docs/general-usage/naming-convention.md:67]: Maybe we should add here that we do not recommend to change the naming convention for an existing dataset and if need to do it, a fresh load to a new dataset would be the least errorprone.

File: dlt/common/normalizers/naming/naming.py [hunk 5]
```diff
@@ -31,15 +39,13 @@ def normalize_table_identifier(self, identifier: str) -> str:
         """Normalizes and shortens identifier that will function as a dataset, table or schema name, defaults to `normalize_identifier`"""
         return self.normalize_identifier(identifier)
 
-    @abstractmethod
     def make_path(self, *identifiers: str) -> str:
```
Comment (sh-rp) [inline dlt/common/normalizers/naming/naming.py:42]: imho these should be called split_path and join_path (same as string operations). not really important but it makes it much clearer what happens here.

File: docs/examples/custom_naming/sql_ci_no_collision.py (hunk 1 above)
Comment (rudolfix) [reply docs/examples/custom_naming/sql_ci_no_collision.py:1]: heh true but people may click in the link and they will go to github. LGTM, for now

File: docs/tools/prepare_examples_tests.py (hunk 3 above)
Comment (rudolfix) [reply docs/tools/prepare_examples_tests.py:60]: something (I think Airflow) creates `logs` folder in the examples folder when I run any example. this prevents me from refreshing examples and I must remove the `logs` folder each time

File: docs/website/docs/general-usage/naming-convention.md (hunk 4 above)
Comment (rudolfix) [reply docs/website/docs/general-usage/naming-convention.md:67]: this documentation needs at least a day of work...

**Pull request merged**